main.tf

```diff
@@ -660,7 +660,7 @@ resource "null_resource" "http_proxy_config_no_proxy_keeper" {
 resource "azapi_update_resource" "aks_cluster_http_proxy_config_no_proxy" {
   count = can(var.http_proxy_config.no_proxy[0]) ? 1 : 0
 
-  type = "Microsoft.ContainerService/managedClusters@2023-01-02-preview"
+  type = "Microsoft.ContainerService/managedClusters@2024-02-01-preview"
```

Comment:
You've assigned two different API for two aks update resources, do we have any special reason for this?